packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts

```diff
@@ -73,7 +72,7 @@ export const useScaffoldContractWrite = <
           () =>
             wagmiContractWrite.writeAsync({
               args: newArgs ?? args,
-              value: newValue ? parseEther(newValue) : value && parseEther(value),
+              value: newValue ? newValue : value && value,
```

Comment:
newValue || value :)

Reply:
lol yup! tysm 🙌

Btw I actually updated it to `??` because checkout this condition : 

```tsx
  const { writeAsync, isLoading } = useScaffoldContractWrite({
    contractName: "YourContract",
    functionName: "setGreeting",
    args: [newGreeting],
    value: parseEther("1"),
  });
  <button onClick={() => writeAsync({ value: parseEther("0") })}>btn</button>;
```

Here we want to respect `0n` and use it instead of the previous value passed to hook